### PR TITLE
remove usages of @Configuration

### PR DESCRIPTION
**Pull request by rexhoffman**
Feel free to mark this up Brian.   Work from a flight so I probably need another set of eyes on this.

### CODE REVIEW

File: AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java [hunk 1]
```diff
@@ -32,20 +32,21 @@
...
  * <ul>
- *  <li>All methods on an @Configuration class are @Bean methods.</li>
- *  <li>The @Configuration class has a public no-parameter constructor.</li>
- *  <li>The @Configuration class must be the top level class in a the java file.</li>
- *  <li>No uses of @ComponentScan on @Configuration classes, instead use @Import</li>
- *  <li>All fields on the @Configuration annotation must be "private static final {Type} {name} = {LITERAL_VALUE}";</li>
+ *  <li>All methods on the class are @Bean methods.</li>
+ *  <li>The class has a public no-parameter constructor.</li>
+ *  <li>The class must be the top level class in a the java file.</li>
+ *  <li>No uses of @ComponentScan on the classes, instead use @Import</li>
```
Comment (toaler) [inline AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java]: @rexhoffman this comment is more related to a @Component not @Bean Lite mode right?

File: AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java [hunk 2]
```diff
@@ -54,26 +55,26 @@
  *  <li>@Bean annotations define at least one name for the bean.</li>
```
Comment (toaler) [inline AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java]: Do you think we should have a bit of context as to why each of these things is important?

File: AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java [hunk 3]
```diff
@@ -54,26 +55,26 @@
...
  *  <li>@Bean method parameters must have a @Qualifier of the bean name they expect as input or an @Value 
  *  of the property they expect spring to inject (system property or other configuration).</li>
  * </ul>
  * <p>
  * With the above restrictions in place, the following checks become feasible, and also occur at compile time:
  * </p>
  * <ul>
- *  <li>Cycles in @Configuration @Import class</li>
+ *  <li>Cycles in @Import class</li>
  *  <li>Cycles in @Bean dependency definitions</li>
```
Comment (toaler) [inline AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java]: What about cycles in @Component?

Comment (rexhoffman) [reply AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java]: Hmm,  not distinguishing about cycles between beans by @Bean or @Component in line 67, though I probably should explicitly mention it.

File: AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java (hunk 2 above)
Comment (rexhoffman) [reply AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java]: I think that is what the readme is for.   I wanted a concise list of checks in the annotation.   Could link out to something though.

**Comment by rahulsh1** [comment]
Can you share details around the need for this PR?

**Comment by rahulsh1** [comment]
There are a couple of @Configuration inside the classes in AptSpringExample

**Comment by rexhoffman** [comment]
The purpose of this pr is to force the use of BeanLite mode, forcing dependencies to be declared through autowired bean definitions on @Bean methods, and save the overhead of creating a proxy at runtime.

**Comment by rexhoffman** [comment]
The example has been updated, though it's value is a little dubious.

File: AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java (hunk 1 above)
Comment (rexhoffman) [reply AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java]: Yes, this entire change list is forcing BeanLite mode when using @Bean definitions.  It is a no-op for standard (non @Bean method containing) @Components.

Comment (toaler) [reply AptSpringAPI/src/main/java/com/salesforce/aptspring/Verified.java]: Ok so we are saying only use @Import and @Component.  If @Bean is needed we can do it from within a @Component class right?  Can I use @Bean in a POJO and then just import via @Import  that POJO to register the lite bean with Spring?

**Pull request merged**